plugin.py

```diff
@@ -14,6 +40,53 @@ class RustAnalyzer(AbstractPlugin):
     def name(cls) -> str:
         return "rust-analyzer"
 
+    @classmethod
+    def basedir(cls) -> str:
+        return os.path.join(cls.storage_path(), "LSP-{}".format(cls.name()))
+
+    @classmethod
+    def server_version(cls) -> str:
+        settings = sublime.load_settings("LSP-rust-analyzer.sublime-settings")
+        return str(settings.get("version"))
+
+    @classmethod
+    def current_server_version(cls) -> str:
+        with open(os.path.join(cls.basedir(), "VERSION"), "r") as fp:
+            return fp.read()
+
+    @classmethod
+    def needs_update_or_installation(cls) -> bool:
+        try:
+            return cls.server_version() != cls.current_server_version()
+        except OSError:
+            return True
+
+    @classmethod
+    def install_or_update(cls) -> None:
+        try:
+            shutil.rmtree(cls.basedir(), ignore_errors=True)
+            os.makedirs(cls.basedir(), exist_ok=True)
+            version = cls.server_version()
+            url = URL.format(tag=version, arch=arch(), platform=platform())
+            gzipfile = os.path.join(cls.basedir(), "server.gz")
+            serverfile = os.path.join(
+                cls.basedir(),
+                "server.exe" if sublime.platform() == "windows" else "server"
+            )
+            with urllib.request.urlopen(url) as fp:
+                with open(gzipfile, "wb") as f:
+                    f.write(fp.read())
+            with gzip.open(gzipfile, "rb") as fp:
+                with open(serverfile, "wb") as f:
+                    f.write(fp.read())
+            os.remove(gzipfile)
+            os.chmod(serverfile, 0o744)
+            with open(os.path.join(cls.basedir(), "VERSION"), "w") as fp:
+                fp.write(version)
+        except Exception:
+            shutil.rmtree(cls.basedir(), ignore_errors=True)
+            raise
```

Comment:
Will this re-raise the original exception automagically?

Reply:
Yes